src/orion/core/worker/primary_algo.py

```diff
@@ -205,3 +316,15 @@ def _verify_trial(self, trial, space=None):
                 f"Trial {trial.id} not contained in space:"
                 f"\nParams: {trial.params}\nSpace: {space}"
             )
+
+
+def _copy_status_and_results(original_trial: Trial, transformed_trial: Trial) -> Trial:
+    """Copies the results, status, and other data from `transformed_trial` to `original_trial`."""
+    transformed_trial = copy.deepcopy(transformed_trial)
```

Comment:
What would you think of doing the opposite?
```
copied_transformed_trial = copy.deepcopy(original_trial)
copied_transformed_trial._params = transformed_trials._params
return copied_transformed_trials
```

Reply:
Ouh, interesting idea! Will try that!

Reply:
Fixed in f728f658f31e39ed7c4ae66f90b83ae4b8b77018